Dev

manualcheg opened the pull request:
Спринт 23
Работа с экраном плейлиста
Версия 1.0

manualcheg committed:
Sprint_23_work_with_playlist
1. Add AppIcon

manualcheg committed:
Sprint_23_work_with_playlist
2. Maked screen to work with playlist with total tracks duration

manualcheg committed:
Sprint_23_work_with_playlist
3. Added field "artworlUrl60" in all Track models

manualcheg committed:
Sprint_23_work_with_playlist
3. RecycleView in PlaylistWorkFragment get tracks. Longtap work, but w/o del track yet.
Fix Dialog buttons color.

manualcheg committed:
Sprint_23_work_with_playlist
3. Delete from playlist work..Not updating recycleView in real-time

manualcheg committed:
Sprint_23_work_with_playlist
3. Cleaning tracks_in_playlist. Not updating recycleView in real-time.

manualcheg committed:
Sprint_23_work_with_playlist
3. Cleaning tracks_in_playlist. Not updating recycleView in real-time.

manualcheg committed:
Sprint_23_work_with_playlist
3. Kostili. Searching solve for update recyclerView in Playlist Screen

manualcheg committed:
Sprint_23_work_with_playlist
3. Solved problem with update recyclerView when delete track
4. Share button is work

manualcheg committed:
Sprint_23_work_with_playlist
4. Delete playlist work

manualcheg committed:
Sprint_23_work_with_playlist
5. Edit playlist screen work

manualcheg committed:
Sprint_23_work_with_playlist
little bugfix

dmitrijegorov commented on app/src/main/res/layout/track_item.xml:
На устройствах с увеличенным размером текста часть обрезается

![Screenshot_20230916_220115](https://github.com/manualcheg/Playlist_maker/assets/46870363/5f26469f-97a2-4d7c-90ea-e077aff1d2f2)
![Screenshot_20230916_220159](https://github.com/manualcheg/Playlist_maker/assets/46870363/b17c4b87-8ce9-477f-97cb-d8901f9a7955)

manualcheg replied:
Исправил

dmitrijegorov commented on app/src/main/java/com/practicum/playlistmaker/mediateka/playlists/data/db/PlaylistDBRepositoryImpl.kt:
Когда-либо добавленные треки остаются в бд независимо от присутствия в плейлистах

<img width="294" alt="Screenshot 2023-09-16 at 22 23 03" src="https://github.com/manualcheg/Playlist_maker/assets/46870363/2e5659eb-59c3-4105-86ca-0e4d7d7cfe59">

manualcheg replied:
Логика проверяющая присутствие трека в других плейлистах есть. 
Для случая удаления трека она работает корректно.
А вот при удалении плейлиста она не применялась. 
Пытаюсь реализовать.

dmitrijegorov commented on app/src/main/java/com/practicum/playlistmaker/mediateka/playlists/data/db/dao/PlaylistsDAO.kt:
```diff
@@ -5,12 +5,22 @@ import androidx.room.Insert
+    @Query("DELETE FROM playlists WHERE playlistId LIKE :playlistId")
+    suspend fun delPlaylist(playlistId: Long)
+
+/*    @Delete(entity = PlaylistEntity::class)
```
Если этот код больше не нужен, то можно смело удалять

manualcheg replied:
Убрал

dmitrijegorov commented on app/src/main/java/com/practicum/playlistmaker/mediateka/playlists/data/db/dao/PlaylistsDAO.kt:
Файл с изображением стоит копировать во внутреннее хранилище


https://github.com/manualcheg/Playlist_maker/assets/46870363/f9ab814f-0dae-4bab-bcc0-5b60faa25914

manualcheg replied:
Исправлено

manualcheg committed:
Sprint_23_work_with_playlist
ver. 1.4 for review
fixed:
• work with big font size
• didn't save cover of playlist to storage
• gray background in placeholder in playlist screen
• didn't hide bottomSheet when click Share in bottomSheetMenu

manualcheg committed:
Sprint_23_work_with_playlist
ver. 1.5 for review
fixed:
• didn't delete tracks when deleting playlist

manualcheg committed:
Sprint_23_work_with_playlist
ver. 1.51 for review
little tuning

manualcheg committed:
Sprint_23_work_with_playlist
ver. 1.51 for review
little tuning 2

manualcheg committed:
Sprint_23_work_with_playlist
ver. 1.52 for review

fix padding in list of playlists on screen for work with playlists
